### ISSUE TITLE
Anchor support

### ISSUE THREAD

**Issue by ElModdy** [vimwiki/vimwiki#666]
Is there a way to set an anchor like the following one?
https://gist.github.com/asabaylus/3071099
I appreciate any kind of help

**Comment by ranebrown** [comment]
Vimwiki supports anchors as noted in the help file `vimwiki-anchors`. For markdown syntax they will be formatted as shown below when using `:VimwikiTOC`
```markdown
# Contents

- [Header1](#Header1)
    - [Header2](#Header1#Header2)
- [Header 1 Spaces](#Header 1 Spaces)
    - [Header 2 Spaces](#Header 1 Spaces#Header 2 Spaces)

# Header1

## Header2

# Header 1 Spaces

## Header 2 Spaces
```

This format isn't compatible with GitHub and GitLab which downcase, drop special characters, replaces spaces with `-` and add a number if necessary to make the link unique. I think it would make sense for Vimwiki to follow this format when using Markdown syntax but it isn't currently supported. If you want to open a PR to modify this behavior that would be great!

**Comment by ElModdy** [comment]
Thanks a lot for the replay, one more quest can I make an anchor that isn't a header?

**Comment by ElModdy** [comment]
Solved just looking at  the doc

**Comment by ranebrown** [comment]
I'm reopening this as an enhancement. I would eventually like to transition to GitHub style anchor links.

**Comment by ranebrown** [comment]
> The code that creates the anchors is here:
> https://github.com/jch/html-pipeline/blob/master/lib/html/pipeline/toc_filter.rb
> 
>     1. It downcases the string
> 
>     2. remove anything that is not a letter, number, space or hyphen (see the source for how Unicode is handled)
> 
>     3. changes any space to a hyphen.
> 
>     4. If that is not unique, add "-1", "-2", "-3",... to make it unique

**Comment by tinmarino** [comment]
Thanks @ranebrown for the [list](https://github.com/vimwiki/vimwiki/issues/666#issuecomment-570753457), From (PR #840) (Issue #831) 1 and 3 are done

Still missing:
- [ ] 2. remove anything that is not a letter, number, space or hyphen (see the source for how Unicode is handled)
- [ ] 4. If that is not unique, add "-1", "-2", "-3",... to make it unique
- [ ] Link generation: #840 only consider link movements

**Comment by tinmarino** [comment]
Fixed in dev: (428c60a: or Search multiple heading with same name)

Related: #968
